src/opm/output/eclipse/Summary.cpp

```diff
@@ -2642,6 +2673,145 @@ namespace Evaluator {
+        using Component  = RateWindow::Component;
+        using Direction  = RateWindow::Direction;
+
+        Opm::EclIO::SummaryNode node_;
+        Opm::UnitSystem::measure m_;
+        std::string regname_{};
+
+        Component component_{ Component::NumComponents };
+        Direction direction_{ Direction::Positive };
+        bool useDirection_{ false };
+        bool isCumulative_{ false };
+        int r1_{ -1 };
+        int r2_{ -1 };
+
+        void analyzeKeyword()
+        {
+            static const auto pattern = std::regex {
+                R"~~(R([OGW])F([RT])([-+GL_]?)([A-Z0-9_]{3})?)~~"
+            };
+
+            auto keywordPieces = std::smatch {};
+            if (std::regex_match(this->node_.keyword, keywordPieces, pattern)) {
+                this->identifyComponent(keywordPieces);
+                this->identifyDirection(keywordPieces);
+                this->identifyCumulative(keywordPieces);
+                this->assignRegionIDs();
+            }
+        }
+
+        double getValue(const RateWindow& iregFlow,
+                        const double      sign,
+                        const double      stepSize) const
+        {
+            const auto val = this->useDirection_
+                ? sign * iregFlow.flow(this->component_, this->direction_)
+                : sign * iregFlow.flow(this->component_);
+
+            return this->isCumulative_
+                ? stepSize * val
+                : val;
+        }
+
+        void assignRegionIDs()
+        {
+            const auto& [r1, r2] =
+                Opm::EclIO::splitSummaryNumber(this->node_.number);
+
+            this->r1_ = r1 - 1;
+            this->r2_ = r2 - 1;
+        }
+
+        void identifyComponent(const std::smatch& keywordPieces)
+        {
+            const auto main = keywordPieces[1].str();
+
+            if (main == "O") {
+                this->component_ = (keywordPieces[3].str() == "G")
+                    ? Component::Vapoil
+                    : Component::Oil;
+            }
+            else if (main == "G") {
+                this->component_ = (keywordPieces[3].str() == "L")
+                    ? Component::Disgas
+                    : Component::Gas;
+            }
+            else if (main == "W") {
+                this->component_ = Component::Water;
+            }
+        }
+
+        void identifyDirection(const std::smatch& keywordPieces)
+        {
+            if (keywordPieces.length(3) == std::smatch::difference_type{0}) {
+                return;
+            }
+
+            const auto dir = keywordPieces[3].str();
+
+            this->useDirection_ = (dir == "+") || (dir == "-");
+
+            if (dir == "-") {
+                this->direction_ = Direction::Negative;
+            }
+        }
+
+        void identifyCumulative(const std::smatch& keywordPieces)
+        {
+            if (keywordPieces.length(2) == std::smatch::difference_type{0}) {
```

Comment:
Seems like this cannot happen for non-optional  `[RT]`.

Reply:
> Seems like this cannot happen for non-optional `[RT]`.

Okay, I'll change to an `assert` then.